phone-to-app-js/client_js.html

```diff
@@ -14,47 +22,84 @@ <h1>Inbound PSTN phone call</h1>
   <button type="button" id="hangup">Hang Up</button>
 
   <script>
-    let token = 'ALICE_JWT';
+    const answerButton = document.getElementById("answer");
+    const rejectButton = document.getElementById("reject");
+    const hangUpButton = document.getElementById("hangup");
+    const notification = document.getElementById("notification");
+    const token = 'ALICE_JWT';
     const client = new vonageClientSDK.VonageClient();
-    const config = new vonageClientSDK.ClientConfig(
-      vonageClientSDK.ConfigRegion.US
-    );
-    client.setConfig(config);
-
     let callId = null;
-    client.createSession(token).then((sessionId) => {
-      console.log(sessionId);
-
-      const answerButton = document.getElementById("answer");
-      const rejectButton = document.getElementById("reject");
-      const hangUpButton = document.getElementById("hangup");
-      const notification = document.getElementById("notification");
-
-      client.on('callInvite', (_callId) => {
-        callId = _callId;
-        notification.textContent = "You are receiving a call";
-        // Answer the call.
-        answerButton.addEventListener("click", () => {
-          client.answerCall(callId);
-          notification.textContent = "You are on a call";
-        });
-        // Reject the call
-        rejectButton.addEventListener("click", () => {
-          client.rejectCall(callId);
-          notification.textContent = `You rejected the call`;
-        });
-        // Hang-up the call
-        hangUpButton.addEventListener("click", () => {
-          client.hangup(callId);
-          notification.textContent = `You ended the call`;
-        });
-      });
 
-      client.on('legStatusUpdate', (callId, legId, status) => {
-        notification.textContent = `Caller Leg Status is: ${status}`;
+    client.createSession(token)
+      .then(sessionId => {
+        console.log("Id of created session: ", sessionId);
+      })
+      .catch(error  => { 
+        console.error("Error creating session: ", error);
       });
-    }).catch((error) => {
-      console.error(error);
+
+    client.on('callInvite', (_callId) => {
+      callId = _callId;
+      notification.textContent = "You are receiving a call";
+      answerButton.style.display = "inline";
+      rejectButton.style.display = "inline";
+    });
+
+    client.on('legStatusUpdate', (_callId, legId, status) => {
+      notification.textContent = `Caller Leg Status is: ${status}`;
+    });
+
+    client.on("callHangup", (_callId, callQuality, reason) => {
```

Comment:
you might wanna use a state enum and one function like `setUIState` that just updates to whatever you set the state to